doc/remailers.rst

```diff
@@ -0,0 +1,125 @@
+
+For compatibility with clients that are not Autocrypt-capable, the remailer
+software might want to offer manual configuration of keys, e.g. via a web
+interface.
+
+Injecting Reamiler Autocrypt-Headers
+~~~~~
+
+To inform subscribers of its key, the remailer injects an Autocrypt-header into
+outgoing messages. The header format is the same as usual, where the ``addr``
+attribute is the address of the list, and ``keydata`` is their Autocrypt key.
+
+In addition to those two attributes, the remailer SHOULD add the
+``_list-recommendation`` attribute, and MAY add ``_list-recipients`` attribute.
+Both of these attributes are exclusive for use by remailers, and MUST NOT be
+used by regular mailbox addresses. Both of these attributes are non-critical
+(starting with an underscore), which will cause an Autocrypt-capable client that
+does not handle these attributes to skip over them, falling back to the usual
+Autocrypt behavior.
+
+For example::
+
+    Autocrypt: addr=post@list.example; _list-recommendation=mandatory; keydata=XXX
+    Autocrypt: addr=post@list.example; _list-recommendation=available; _list-recipients=7/15; keydata=XXX
+
+The ``_list-recommendation`` attribute SHOULD be included in an Autocrypt header
+sent by a remailer. Its value may be either ``available`` or ``mandatory``. If
+the value is set to ``mandatory``, the remailer requires encryption of every
+message, and will reject plaintext messages. If the value is ``available``,
+encryption is offered as an opt-in mechanism and plaintext messages will be
+handled as usual. In either case, encrypted messages will be delivered
+exclusively to subscribers for whom an encryption key is available.
+
+To inform the decision of whether to encrypt or not, the remailer MAY include a
+``_list-recipients`` attribute. This attribute contains the current number of
+subscribers for which an encryption key is available, and the total number of
+subscribers, separated by a slash character ('/'). This attribute is only useful
+if the ``_list-recommendation`` attribute is set to ``available``.
+
+Administrative Considerations
+----
+
+Graceful Up- and Downgrade
+~~~
+
+To allow graceful up- and downgrade of lists with no intervention by the list
+administrator, a remailer software distribution could automatically include
+Autocrypt-headers with outgoing mail depending on the number of
+Autocrypt-enabled subscribers. For example, a remailer software distribution may
+choose to inject Autocrypt headers only if encryption keys are available for a
+majority of subscribers. Note that obviously, end-to-end encryption is
+unsuitable for lists that are publicly archived.
+
+Missed Message Notifications
+~~~
+
+A remailer MAY send notifications to recipients that were unable to receive an
+encrypted message, if their key was unavailable or unusable for some reason. If
+available, a digest mechanism can be used to aggregate this information and
+avoid redundant notifications.
+
```

Comment:
I like this suggestion about notifications, but i don't think it goes far enough.

It should also discuss the remailer sending notifications to the message *sender* that some number of recipients were unable to receive a given cleartext message?

Reply:
This can get annoying fast. I also think that if the client is autocrypt-mailing-list-capable, the `_list-recipients` attribute is a much nicer way to achieve this a priori. A notification to the sneder would be more of a fallback for non-autocrypt clients I guess?

Reply:
well, _list-recipients is what's true at *last message recieved* time, but during actual new message delivery, the state might be different.

"this can get annoying fast" is probably true, but it's also true for the non-crypto recipients.  we can mitigate that problem with the same "digest mechanism" idea, right?

Reply:
I think last message receive time is fairly good info on mailing lists that aren't very inactive, and the rough impression will still be right. Having both mechanisms seems redundant, and if I had to choose the header seems preferable.

> "this can get annoying fast" is probably true, but it's also true for the non-crypto recipients. we can mitigate that problem with the same "digest mechanism" idea, right?

If we wanted to go this route, a limit of one message every X hours/days would make more sense, for immediate feedback when a message is sent, without being spammy for multiple messages (and the info is not going to change much between batches sent).